pyomo/contrib/mindtpy/mip_solve.py

```diff
@@ -2,17 +2,280 @@
 from __future__ import division
 
 from pyomo.contrib.gdpopt.util import copy_var_list_values
-from pyomo.core import Constraint, Expression, Objective, minimize, value
+from pyomo.core import Constraint, Expression, Objective, minimize, value, Var
 from pyomo.opt import TerminationCondition as tc
 from pyomo.opt import SolutionStatus, SolverFactory
 from pyomo.contrib.gdpopt.util import SuppressInfeasibleWarning, _DoNothing
 from pyomo.contrib.gdpopt.mip_solve import distinguish_mip_infeasible_or_unbounded
+from pyomo.solvers.plugins.solvers.persistent_solver import PersistentSolver
+
+from pyomo.contrib.mindtpy.nlp_solve import (solve_NLP_subproblem,
+                                             handle_NLP_subproblem_optimal, handle_NLP_subproblem_infeasible,
+                                             handle_NLP_subproblem_other_termination, solve_NLP_feas)
+from pyomo.contrib.mindtpy.cut_generation import (add_oa_cuts,
+                                                  add_int_cut)
+from pyomo.contrib.gdpopt.util import copy_var_list_values, identify_variables
+from math import copysign
+from pyomo.environ import *
+from pyomo.core import Constraint, minimize, value
+from pyomo.core.expr import current as EXPR
+from math import fabs
+
+from pyomo.repn import generate_standard_repn
+
+try:
+    import cplex
+    from cplex.callbacks import LazyConstraintCallback
+except ImportError:
+    print("Cplex python API is not found. Therefore, lp-nlp is not supported")
+    '''Other solvers (e.g. Gurobi) are not supported yet'''
+
+
+class LazyOACallback(LazyConstraintCallback):
```

Comment:
As this class is hard-coded to the CPLEX API, I would recommend including 'cplex' in the class name.  Maybe `LazyOACallback_cplex`?  That hints at what you would need to change/reimplement to add, e.g., Gurobi support.

Reply:
Thanks for the advice. `LazyOACallback` has been changed into `LazyOACallback_cplex` in the latest commit.